sw/device/lib/meson.build

```diff
@@ -17,6 +17,7 @@ sw_lib_uart = declare_dependency(
       sw_lib_base_print,
       sw_lib_runtime_ibex,
       dif_uart,
+      top_earlgrey,
```

Comment:
Are these dependencies needed, given we are only using the macro defined in `top_earlgrey.h`?

Reply:
`top_earlgrey.h` has some declarations for data that is defined in `top_earlgrey.c` - in order to avoid linker definition errors in future, it has always been our policy to link the C files for any included headers.

Are you running into a problem because of this dependency?